File: benchmarks/sieve/sieve.pl
use strict;
use warnings;

my $n = 1000000;
my @prime = ();

for (my $i = 0; $i <= $n; $i++) {
    push @prime, 1;
}

$prime[0] = 0;
$prime[1] = 0;

for (my $i = 2; $i * $i <= $n; $i++) {
    if ($prime[$i]) {
        for (my $j = $i * $i; $j <= $n; $j += $i) {
            $prime[$j] = 0;
        }
    }
}

my $count = 0;

for (my $i = 0; $i <= $n; $i++) {
    $count++ if $prime[$i];
}

print "$count\n";

File: benchmarks/sieve/sieve.rb
n = 1000000
prime = []

for i in 0..n
  prime.push(true)
end

prime[0] = false
prime[1] = false

i = 2
while i * i <= n
  if prime[i]
    j = i * i
    while j <= n
      prime[j] = false
      j += i
    end
  end
  i += 1
end

count = 0

for i in 0..n
  count += 1 if prime[i]
end

puts count